pkg/activator/handler/concurrency_reporter.go

```diff
@@ -136,7 +145,11 @@ func (cr *ConcurrencyReporter) report(now time.Time) []asmetrics.StatMessage {
 		cr.mux.Lock()
 		defer cr.mux.Unlock()
 		for _, key := range toDelete {
-			delete(cr.stats, key)
+			// Avoid deleting the stat if a request raced fetching it while we've been
+			// busy reporting.
+			if cr.stats[key].refs.Load() == 0 {
```

Comment:
Is there a case to worry about here where the ref is > 1 but we still don't want to delete? I'm thinking we got it normally (so refs = 1) for a couple cycles, then computeReport decides to delete it because concurrency went 0 but then another request comes in in the gap before we take the lock again (so we should no longer delete it) - is that a thing?

Reply:
The refs are all incremented under at least a read lock and read under a write lock, so I'd think they are consistent.

The gap you describe is essentially the gap this is aiming to fix and that exists today.

Reply:
oh wait, I wasn't following the logic correctly at all - I guess the coffee hasn't quite landed yet this morning. The Incs() have a matching Dec() so we're doing proper reference counting essentially. How do the benchmarks compare before/after this?

Reply:
Added benchmark results to the main description.